File: 899-binary-gap/binary-gap.cs
public class Solution {
    public int BinaryGap(int n) {
        int ans = 0, pos = 0, last = -1;

        while (n > 0) {
            if ((n & 1) == 1) {
                if (last != -1) {
                    ans = Math.Max(ans, pos - last);
                }
                last = pos;
            }
            n >>= 1;
            pos++;
        }

        return ans;
    }
}

File: 899-binary-gap/binary-gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        b = bin(n)[2:]
        x, mx = 0, 0

        for i in range(1, len(b)):
            if b[i] == "1":
                mx = max(mx, i - x)
                x = i

        return mx
